redwood-treehouse/src/androidMain/kotlin/app/cash/redwood/treehouse/hotReloadAndroid.kt

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package app.cash.redwood.treehouse
+
+import java.net.URI
+import java.util.concurrent.TimeUnit
+import kotlin.time.Duration.Companion.milliseconds
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.channels.trySendBlocking
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.transformLatest
+import kotlinx.coroutines.launch
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import okhttp3.WebSocket
+import okhttp3.WebSocketListener
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal actual fun hotReloadFlow(flow: Flow<String>): Flow<String> {
+  return flow.transformLatest { manifestUrl ->
+    val wsClient: OkHttpClient = OkHttpClient.Builder()
+      .readTimeout(0, TimeUnit.MILLISECONDS)
+      .connectTimeout(1, TimeUnit.SECONDS)
+      .build()
```

Comment:
A new client is being created for every upstream emission when what you probably want is to share this instance for the lifetime of the downstream subscription. You can do that with a level of nesting such as
```kotlin
return flow {
  val client = OkHttpClient.Builder()...build()

  emitAll(upstream.flatMapLatest { logic... })
}
```